packages/blade/src/components/Collapsible/CollapsiblePanel.web.tsx

```diff
@@ -0,0 +1,127 @@
+import type { ReactElement, ReactNode, TransitionEventHandler } from 'react';
+import { useRef } from 'react';
+import styled from 'styled-components';
+import { useCollapsibleContext } from './CollapsibleContext';
+import { castWebType, makeMotionTime, makeSize } from '~utils';
+import { useDidUpdate } from '~src/hooks/useDidUpdate';
+
+type CollapsiblePanelProps = {
+  children: ReactNode;
+};
+
+type StyledCollapsiblePanelProps = {
+  defaultIsExpanded: boolean;
+  isExpanded: boolean;
+};
+
+/**
+ * We can't animate to and from `auto` height, these are used for imperative css transitions.
+ * Overall on expanding height will change as: 0px -> Actual height -> auto
+ */
+const HEIGHT_EXPANDED = 'auto';
```

Comment:
why do we need to go from 0 ➡️ actual ➡️ auto and not 0 ➡️ actual ➡️ 0?

Reply:
hmm I see, then shouldn't it just be 0 ➡️ actual? my main point is what is the significance of auto here?

Reply:
okay got my answer [here](https://github.com/razorpay/blade/pull/1262#discussion_r1219602799)